src/js/components/timeline.js

```diff
@@ -1,75 +1,33 @@
 import React from 'react';
-import TweetBlock from './tweetBlock';
-import {promiseTimeline} from '../services/timelineService';
+import TweetsContainer from './tweetsContainer';
 
 export default class Timeline extends React.Component {
     constructor(props){
         super(props);
-        this.state = {
-            tweets: [], // tweets retrieved 
-            isLoaded: false, // flags if request was loaded or not
-            error: null, // error message
-        };
-        this.errorMessage = "Something went wrong, please contact systems administrator.";
-    }
-
-    componentDidMount() { // calls requestTimeline upon component being mounted
-        this.requestTimeline();
+        if(this.props?.type){
```

Comment:
I've never seen this before, how does this work with the question mark there? Also, it's probably not necessary to attach the value of className the the instance of Timeline. You could always declare a local variable later in the render method